src/metabase/models/setting.clj

```diff
@@ -1342,3 +1343,42 @@
            (map (fn [[setting-name]]
                   [setting-name (get setting-name)])))
      @registered-settings)))
+
+(defn- set-via-env? [setting]
+  (contains? env/env (setting-env-map-name setting)))
+
+(defmulti ^:private validate-setting-format :type)
+
+;; Treat setting as valid unless a validator has been defined for its type
+(defmethod validate-setting-format :default [_] nil)
+
+(defmethod validate-setting-format :json [{:keys [name]}]
+  (try
+    ;; attempt to parse the string to trigger any related exception
+    (doall (json/parse-string (get-value-of-type :string name)))
+    ;; there was no issue parsing the setting, so return a falsey value
+    nil
+    (catch JsonParseException e e)
+    ;; be conservative about what is returned, to avoid accidentally exposing secrets in the value, so do
+    (catch Exception e (ex-info "Unable to validate setting" {:exception-type (type e)}))))
```

Comment:
There's no test for this case. I'm actually wondering whether it's worth keeping around. Can you justify it?